resources/measures/nighttime_oa_controls/measure.rb

```diff
@@ -0,0 +1,200 @@
+  def self.air_loop_hvac_unitary_system?(air_loop_hvac)
+    is_unitary_system = false
+    air_loop_hvac.supplyComponents.each do |component|
+      obj_type = component.iddObjectType.valueName.to_s
+      case obj_type
+      when 'OS_AirLoopHVAC_UnitarySystem', 'OS_AirLoopHVAC_UnitaryHeatPump_AirToAir', 'OS_AirLoopHVAC_UnitaryHeatPump_AirToAir_MultiSpeed', 'OS_AirLoopHVAC_UnitaryHeatCool_VAVChangeoverBypass'
+        is_unitary_system = true
+      end
+    end
+    return is_unitary_system
+  end
+
+  # determine if the air loop is residential (checks to see if there is outdoor air system object)
+  def self.air_loop_res?(air_loop_hvac)
+    is_res_system = true
+    air_loop_hvac.supplyComponents.each do |component|
+      obj_type = component.iddObjectType.valueName.to_s
+      case obj_type
+      when 'OS_AirLoopHVAC_OutdoorAirSystem'
+        is_res_system = false
+      end
+    end
+    return is_res_system
+  end
+
+  # Determine if the system is a DOAS based on
+  # whether there is 100% OA in heating and cooling sizing.
+  def self.air_loop_doas?(air_loop_hvac)
+    is_doas = false
+    sizing_system = air_loop_hvac.sizingSystem
+    if sizing_system.allOutdoorAirinCooling && sizing_system.allOutdoorAirinHeating && (NighttimeOAControls.air_loop_res?(air_loop_hvac) == false) && (air_loop_hvac.name.to_s.include?("DOAS") || air_loop_hvac.name.to_s.include?("doas"))
+      is_doas = true
+    end
+    return is_doas
+  end
+
+  # Determine if is evaporative cooler
+  def self.air_loop_evaporative_cooler?(air_loop_hvac)
+    is_evap = false
+    air_loop_hvac.supplyComponents.each do |component|
+      obj_type = component.iddObjectType.valueName.to_s
+      case obj_type
+      when 'OS_EvaporativeCooler_Direct_ResearchSpecial', 'OS_EvaporativeCooler_Indirect_ResearchSpecial', 'OS_EvaporativeFluidCooler_SingleSpeed', 'OS_EvaporativeFluidCooler_TwoSpeed'
+        is_evap = true
+      end
+    end
+    return is_evap
+  end
+
+  # define what happens when the measure is run
+  def run(model, runner, user_arguments)
+    super(model, runner, user_arguments)  # Do **NOT** remove this line
+
+    # use the built-in error checking
+    if !runner.validateUserArguments(arguments(model), user_arguments)
+      return false
+    end
+
+	
+	template = 'ComStock 90.1-2013'
+    std = Standard.build(template)
+
+    # loop through air loops
+    unitary_system_count = 0
+    li_unitary_systems = []
+    non_unitary_system_count = 0
+    li_non_unitary_systems = []
+    model.getAirLoopHVACs.sort.each do |air_loop_hvac|
+      # skip systems that are residential, or are DOAS
+      next if NighttimeOAControls.air_loop_res?(air_loop_hvac)
+      next if NighttimeOAControls.air_loop_doas?(air_loop_hvac)
+      # skip outpatient healthcare and schools 
+      next if ['outpatient', 'Outpatient', 'OUTPATIENT', 'school', 'SCHOOL', 'School', 'k12', 'K12', 'education', 'EDUCATION', 'Education'].any? { |word| (air_loop_hvac.name.get).include?(word) }
```

Comment:
We should probably remove small and large hotels too. So I guess this measure is most suitable for retail, offices, and warehouses.

Reply:
Done--thanks!